src/Git/Git.h

```diff
@@ -35,11 +35,32 @@ struct git_repository;
 class CFilterData
 {
 public:
+
+	enum
+	{
+		SHOW_LAST_N_COMMITS = 0,
+		SHOW_LAST_SELECTED_DATE = 1,
+		SHOW_LAST_N_YEARS = 2,
+		SHOW_LAST_N_MONTHS = 3,
+		SHOW_LAST_N_WEEKS = 4,
+	};
+
 	CFilterData()
 	{
 		m_From=m_To=-1;
 		m_IsRegex=1;
+		m_NumberOfLogsScale = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfLogsScale"), -1);
+		m_NumberOfLogs = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfLogs"), 0);
+		m_NumberOfYears = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfYears"), 0);
+		m_NumberOfMonths = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfMonths"), 0);
+		m_NumberOfWeeks = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfWeeks"), 0);

```

Comment:
@csware 
I found some code in D:\TortoiseGit\src\TortoiseProc\Commands\LogCommand.cpp, 
probably from TSvn:
```
bool LogCommand::Execute()
{
	...
	CString val = parser.GetVal(_T("limit"));
	int limit = _tstoi(val);
	...
	if (limit == 0)
	{
		CRegDWORD reg = CRegDWORD(_T("Software\\TortoiseGit\\NumberOfLogs"), 100);
		limit = (int)(LONG)reg;
	}
	...
	CLogDlg dlg;
	theApp.m_pMainWnd = &dlg;
	dlg.SetParams(orgCmdLinePath, cmdLinePath, rev, range, limit);
```


3 questions:

1. The ```limit``` option is there, but limit(```CLogDlg::m_limit```) is not used any more. Fix it or drop it?
2. If fix limit option, use ```Software\TortoiseGit\LogDialog\``` or keep ```Software\TortoiseGit\```???
3. If fix limit option, add one more ```scale``` option for LogCommand?

My thoughts for those 3 questions:

1. fix limit option, because it can be done in this PR.
2. keep ```Software\TortoiseGit\``` for upstream compatibility reason.
3. no need scale option. Also for upstream compatibility reason. Once the limit option was given, uses the last commits of scale. (But, I can imagine some users may fire a issue to add it in the future.)